dev/AppNotifications/AppNotifications.idl

```diff
@@ -111,6 +111,29 @@ namespace Microsoft.Windows.AppNotifications
 
         // Gets or sets whether a Notification's pop-up UI is displayed on the user's screen.
         Boolean SuppressDisplay;
+
+        // Gets or sets the Notification Device Data
+        AppNotificationDevicesData DevicesData;
+    }
+
+    // The Notification Device Data
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass AppNotificationDevicesData
+    {
+        // Initializes a new Instance of NotificationDevicesData
+        AppNotificationDevicesData();
+
+        // Checks if Video or Audio Calling is supported
+        static Boolean IsVideoOrAudioCallingSupported();
+
+        // Gets or sets the Camera Device Id
+        String CameraDeviceId{ get; set; };
```

Comment:
If this is set to the empty string (note: `String` cannot be null) or an invalid device identifier, I presume your toast will not show anything?

Reply:
Toast will show system default device id as the ticked one if it's null / empty
![image](https://github.com/user-attachments/assets/f064371c-ca53-4291-81bf-f0a62461f419)

Reply:
Device validation is performed within the shell code. First, the shell invokes the existing enumerate API to retrieve a list of connected devices. It then compares these devices with those provided by the aforementioned APIs. If a match is found, the device is selected for preview and selected in the list (as shown in last mock image above). Otherwise, 'system default' devices will be displayed for preview and selection.